MyCryptoHQ/MyCrypto
# Mnemonic Redesign

Closes #1279

### Description

Daniel T. noted the inherent insecurity of typing the the mnemonic words after revelation, and suggested a move to a shuffled, click-based system to avoid keylogging.

### Changes

*Before*: User generates a new wallet via mnemonic phrase. In the confirmation stage, each word is entered via keyboard in order prior to proceeding.

*Now*: User generates a new wallet via mnemonic phrase. In the confirmation stage, the words are shuffled and turned into buttons, and the user must click them in the correct order.

### Steps to Test

1. From the home page, click "Create New Wallet".
2. Select the option for "Generate a Mnemonic Phrase".
3. Select the option for 'Confirm Phrase'.
4. Select the correct first word. It should flash green, become disabled, and show an index representing its place.
5. Select an incorrect next word. It should flash red, indicating that it was not the correct choice.
6. Press the 'Reveal Next Word' button. The word that is next in sequence should flash green, indicating it is the next valid choice.
7. Select all words and ensure the created wallet is accessible.

### Screenshots

_Fresh screen_
<img width="825" alt="fresh-confirming" src="https://user-images.githubusercontent.com/10479826/38700540-3e11da58-3e50-11e8-9a4f-b3e14670e092.png">

_Flashing an error on incorrect click_
<img width="820" alt="flashing-error" src="https://user-images.githubusercontent.com/10479826/38700551-4a9c910a-3e50-11e8-95c6-e6020de1d437.png">

_Flashing when being revealed_
<img width="812" alt="being-revealed" src="https://user-images.githubusercontent.com/10479826/38700561-540bf000-3e50-11e8-9f50-2d35ea949158.png">

_Successful selection_
<img width="1290" alt="screen shot 2018-04-12 at 2 26 36 pm" src="https://user-images.githubusercontent.com/10479826/38705196-a67f22b4-3e5d-11e8-8510-3cc1ab2c2302.png">
